dockstore-client/src/main/java/io/github/collaboratory/cwl/CWLClient.java

```diff
@@ -1040,8 +1040,14 @@ private void writeJob(String jobOutputPath, org.json.simple.JSONObject newJson)
             } else {
                 assert (files.size() == 1);
                 FileProvisioning.FileInfo file = files.get(0);
-                final Map<String, Object> fileMapDataStructure = (Map)(outputObject).get(key);
-                outputSet.addAll(provisionOutputFile(key, file, fileMapDataStructure));
+                Object o = outputObject.get(key);
+                if (o instanceof Map) {
+                    final Map<String, Object> fileMapDataStructure = (Map)(outputObject).get(key);
+                    outputSet.addAll(provisionOutputFile(key, file, fileMapDataStructure));
+                }
+                if (o instanceof Integer) {
```

Comment:
Why would it be an integer and what else can it be? Some comments about the expectations would be nice.

Reply:
It's a new test case that we haven't handled before.  Previously it was always a map and nothing else (as seen from the statements above it).  I don't know what else it can be.

Reply:
What is the value of the integer? Is it a 0 on success, nonzero when not? 
But then, would a failing workflow ever get here and would this be guaranteed to be 0?
(Answer in javadoc preferably)